packages/forms/resources/views/components/checkbox-list.blade.php

```diff
@@ -58,15 +58,23 @@
             },
 
             toggleAllCheckboxes: function () {
+                updatedStates = []
                 state = ! this.areAllCheckboxesChecked
 
                 this.visibleCheckboxListOptions.forEach((checkboxLabel) => {
                     checkbox = checkboxLabel.querySelector('input[type=checkbox]')
 
                     checkbox.checked = state
-                    checkbox.dispatchEvent(new Event('change'))
+
+                    if (state) {
+                        updatedStates.push(checkbox.value);
+                    }
+
+                    this.checkIfAllCheckboxesAreChecked()
```

Comment:
We probably should move this outside of the loop, so it only happens once afterwards?

Reply:
Done